test-graalvm-hibernate-jpa/build.gradle

```diff
@@ -0,0 +1,114 @@
+plugins {
+    id 'io.micronaut.application' apply false
+}
+
+def sharedMainJava = file('src/main/java')
+def sharedMainResources = file('src/main/resources')
+def sharedTestGroovy = file('src/test/groovy')
+
+subprojects {
+
+    apply plugin: 'io.micronaut.application'
+    apply plugin: "java"
+    apply plugin: "groovy"
+
+    group "io.micronaut.example"
+
+    repositories {
+        jcenter()
+        mavenCentral()
+    }
+
+    micronaut {
+        version project.ext.micronautVersion
+        runtime "netty"
+        testRuntime "spock"
+    }
+
+    sourceSets {
+        main {
+            java {
+                srcDirs sharedMainJava
+            }
+            resources {
+                srcDirs sharedMainResources
+            }
+        }
+        test {
+            groovy {
+                srcDirs sharedTestGroovy
+            }
+        }
+    }
+
+    sourceCompatibility = 1.8
+    targetCompatibility = 1.8
+
+    dependencies {
+        runtimeOnly "io.micronaut.sql:micronaut-jdbc-hikari"
+        runtimeOnly "ch.qos.logback:logback-classic"
+
+        implementation "io.micronaut:micronaut-http-client"
+        implementation project(":hibernate-jpa")
+
+        testImplementation "io.micronaut.test:micronaut-test-core:$micronautTestVersion"
+        testImplementation platform("org.testcontainers:testcontainers-bom:$testContainersVersion")
+        testImplementation "org.testcontainers:spock"
+        testImplementation "org.testcontainers:jdbc"
+    }
+
+    nativeImage {
+        args("-H:+TraceClassInitialization")
+        args("-H:IncludeResources=logback.xml|application.yml")
```

Comment:
This is shouldn't be necessary. The Graal type element visitor includes automatically all resources.

Reply:
Somehow it doesn't work, you can try to remove it and run `./gradlew :test-graalvm-hibernate-jpa:h2:testNativeImage`

Reply:
I see it now.

This is the output of an application that works:

```
> Task :compileJava
Note: Writing native-image.properties file to destination: META-INF/native-image/example.micronaut/basic-app/native-image.properties
Note: Writing reflection-config.json file to destination: META-INF/native-image/example.micronaut/basic-app/reflection-config.json
Note: Writing resource-config.json file to destination: META-INF/native-image/example.micronaut/basic-app/resource-config.json
```

Note that the `resource-config.json` is created.


But for this project, the file is not created:

```
> Task :test-graalvm-hibernate-jpa:h2:compileJava
Note: Writing native-image.properties file to destination: META-INF/native-image/io.micronaut.example/h2/native-image.properties
Note: Writing reflection-config.json file to destination: META-INF/native-image/io.micronaut.example/h2/reflection-config.json
Note: Creating bean classes for 6 type elements
```

I can't debug the type element visitor to see why, it never stops on my breaking point. I suspect that maybe it's because of the multiproject build can't find the `resources` directory: https://github.com/micronaut-projects/micronaut-core/blob/69ec24c34aebc244209f4e2ced47af694a4d05a1/graal/src/main/java/io/micronaut/graal/reflect/GraalTypeElementVisitor.java#L323

Reply:
Looks like the code check for the resources dir in the current directory, but in this case, it's in a different location.

Reply:
I see :+1: